pkgs/applications/window-managers/wmfs/default.nix

```diff
@@ -0,0 +1,37 @@
+{ stdenv, fetchFromGitHub, gnumake,
+  libX11, libXinerama, libXrandr, libXpm, imlib2 }:
+stdenv.mkDerivation rec
+{
+  name = "wmfs-${version}";
+  version = "201902";
+  src = fetchFromGitHub {
+    owner = "xorg62";
+    repo = "wmfs";
+    sha256 = "sha256:1m7dsmmlhq2qipim659cp9aqlriz1cwrrgspl8baa5pncln0gd5c";
+    rev = "b7b8ff812d28c79cb22a73db2739989996fdc6c2";
+  };
+  nativeBuildInputs = [
+    gnumake
+  ];
+  buildInputs = [
+    imlib2
+    libX11
+    libXinerama
+    libXrandr
+    libXpm
+  ];
+  # XFT doesn't quite work for some reason
+  configurePhase = ''
+    ./configure \
+    --without-xft \
+    --prefix $out \
+    --xdg-config-dir $out/etc \
+    --man-prefix $out/share
+  '';
```

Comment:
you can probably replace by
```
configureFlags = [ "--without-xft" "--xdg-config-dir=${placeholder "out"}" "--man-prefix=${placeholder "out"}/share" ];
```
Then overrides can use preConfigure and postConfigure.

Reply:
I tried to do exactly that initially, but it didn't work for some reason.
It shows:
`LDFLAGS= -lX11 -lXinerama -lImlib2 -lxft -lfreetype`
and then
```
src/wmfs.h:26:10: fatal error: X11/Xft/Xft.h: No such file or directory
 #include <X11/Xft/Xft.h>
          ^~~~~~~~~~~~~~~
compilation terminated.
```
Doesn't happen when done with `configurePhase`. I don't know if this is a bug or I am doing something wrong.

Reply:
ok here is how I got it to work:
```diff
diff --git a/pkgs/applications/window-managers/wmfs/default.nix b/pkgs/applications/window-managers/wmfs/default.nix
index 97a45fe59ad..03c5a39ea4d 100644
--- a/pkgs/applications/window-managers/wmfs/default.nix
+++ b/pkgs/applications/window-managers/wmfs/default.nix
@@ -1,5 +1,5 @@
 { stdenv, fetchFromGitHub, gnumake,
-  libX11, libXinerama, libXrandr, libXpm, imlib2 }:
+  libX11, libXinerama, libXrandr, libXpm, libXft, imlib2 }:
 stdenv.mkDerivation rec
 {
   name = "wmfs-${version}";
@@ -19,15 +19,16 @@ stdenv.mkDerivation rec
     libXinerama
     libXrandr
     libXpm
+    libXft
   ];
-  # XFT doesn't quite work for some reason
-  configurePhase = ''
-    ./configure \
-    --without-xft \
-    --prefix $out \
-    --xdg-config-dir $out/etc \
-    --man-prefix $out/share
+  preConfigure = ''
+    sed -i s/-lxft/-lXft/g configure
   '';
+  makeFlags = [
+    "PREFIX=${placeholder "out"}"
+    "XDG_CONFIG_DIR=${placeholder "out"}/etc/xdg"
+    "MANPREFIX=${placeholder "out"}/share/man"
+  ];
   meta = {
     description = "Window manager from scratch";
     license = stdenv.lib.licenses.bsd2;
```

Reply:
Oh, that makes perfect sense actually. I'm going to do that, thanks.